download_data/about_sitemap/download_all_pdf_url.py

```diff
@@ -0,0 +1,107 @@
+import urllib.request
+import gzip
+import io
+from lxml import etree
+import argparse
+import json
+import re
+from pathlib import Path
+from tqdm import tqdm 
+
+def get_sitemap_text(gz_url):
+    """
+    Args:
+        sitemap url (URL links with .gz suffix)
+
+    Returns:
+        Sitemap text
+    """
+    response = urllib.request.urlopen(gz_url)
+    compressed_file = io.BytesIO(response.read())
+    decompressed_file = gzip.GzipFile(fileobj=compressed_file)
+    return decompressed_file.read()
+
+
+def parse_all_son_sitemap_url(sitemap_text):
+    """Parse all son sitemap link in root sitemap file text
+
+    Returns:
+        Sitemap link list
+    """
+    root = etree.fromstring(sitemap_text)
+    locs = []
+    for sitemap in root.findall('{http://www.sitemaps.org/schemas/sitemap/0.9}sitemap'):
+        loc = sitemap.find('{http://www.sitemaps.org/schemas/sitemap/0.9}loc').text
+        locs.append(loc)
+    return locs
+
+
+def get_pdf_url_in_sitemap_text(sitemap_text):
+    """Get all pdf links in the son sitemap file text
+
+    Returns:
+        Pdf link list
+    """
+    root = etree.fromstring(sitemap_text)
+    locs = []
+    for url in root.findall('{http://www.sitemaps.org/schemas/sitemap/0.9}url'):
+        loc = url.find('{http://www.sitemaps.org/schemas/sitemap/0.9}loc').text
+        locs.append(loc)
+    return locs
+
+
+LANG_LIST = ["ar", "en", "es", "fr", "ru", "zh"]
+
+LANGUAGE_REGEX = re.compile(f"({'|'.join(LANG_LIST)})\.pdf$")
+
+def match_six_countries_file_url(url_list):
+    """Find a matching PDF link for the six languages in a URL linked list
+
+    Returns:
+        PDF links in well matched six languages
+
+        Examples:
+            [["https://xxx-zh.pdf", "https://xxx-en.pdf", ...], ...]
```

Comment:
TODO: change this function to `def analyse_url`
example:
https://digitallibrary.un.org/record/4008528/files/E_2023_SR.7-EN.pdf ->
```
{"record": 4008528, "lang": "en", "type": "pdf"}
```